### ISSUE TITLE
es6 + async/await + babel

### ISSUE THREAD

**Issue by ronag** [deepstreamIO/deepstream.io#400]
From a coding perspective I think a lot of the code could be simplified (with improved correctness!) by going the es6 + async/await with babel route. Especially in terms of asynchronous stuff.

async/await has already landed in node 7 nightly...

Is there any reason for compatibility with older node versions?

**Comment by yasserf** [comment]
You make a good point in terms of it simplifying code. We do want to support LTS versions though, currently just moved to using v6. We will consider this moving forward as we move more stuff into C land / v2.0, but that is still a while away.

**Comment by ronag** [comment]
Hm... not sure if I'm to thrilled about C land. What parts are you looking into moving there?

**Comment by ronag** [comment]
You can still support LTS versions through babel.

**Comment by yasserf** [comment]
I guess I'll reword it to that I'm not very keen on transpiling. If we end up using async we'll probably do it via co-routines in bluebird and switch to native once its supported.

We are moving all connection related logic to reduce the overhead of things being called from node into c as often. We already decided on only using uws and are moving our parsing and distribution logic first.
